crates/networking/rpc/rpc.rs

```diff
@@ -69,6 +69,80 @@ use tower_http::cors::CorsLayer;
 use tracing::{error, info};
 use tracing_subscriber::{EnvFilter, Registry, reload};
 
+#[cfg(feature = "jemalloc_profiling")]
+use axum::response::IntoResponse;
+
+#[cfg(feature = "jemalloc_profiling")]
+pub async fn handle_get_heap() -> Result<impl IntoResponse, (StatusCode, String)> {
+    let Some(mutex) = jemalloc_pprof::PROF_CTL.as_ref() else {
+        return Err((
+            StatusCode::NOT_IMPLEMENTED,
+            "jemalloc profiling is not available (compile with `jemalloc_profiling` and make sure env_target is not msvc)".into(),
```

Comment:
We should mention somewhere (maybe in this message) that this feature only works on Linux.